Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/Pointers/PokePointer.cs

```diff
@@ -177,52 +172,66 @@ private void TryRaisePokeDown(GameObject targetObject, Vector3 touchPosition)
             }
         }
 
-        private void TryRaisePokeUp(Vector3 touchPosition)
+        private void TryRaisePokeUp(GameObject targetObject, Vector3 touchPosition)
         {
-            if (currentTouchableDown != null)
+            if (currentTouchableObjectDown != null)
             {
-                if (currentTouchableDown.EventsToReceive == TouchableEventType.Pointer)
+                Debug.Assert(Result.CurrentPointerTarget == currentTouchableObjectDown, "PokeUp will not be raised for correct object.");
+
+                if (closestProximityTouchable.EventsToReceive == TouchableEventType.Pointer)
                 {
                     MixedRealityToolkit.InputSystem?.RaisePointerUp(this, pointerAction, Handedness);
                 }
-                else if (currentTouchableDown.EventsToReceive == TouchableEventType.Touch)
+                else if (closestProximityTouchable.EventsToReceive == TouchableEventType.Touch)
                 {
                     MixedRealityToolkit.InputSystem?.RaiseOnTouchCompleted(InputSourceParent, Controller, Handedness, touchPosition);
                 }
 
-                currentTouchableDown = null;
+                currentTouchableObjectDown = null;
             }
         }
 
         private void RaiseTouchUpdated(GameObject targetObject, Vector3 touchPosition)
         {
-            if (currentTouchableDown != null)
+            if (currentTouchableObjectDown != null)
             {
-                Debug.Assert(currentTouchableDown.gameObject == targetObject);
-                if (currentTouchableDown.EventsToReceive == TouchableEventType.Touch)
+                Debug.Assert(Result?.CurrentPointerTarget == currentTouchableObjectDown);
+
+                if (closestProximityTouchable.EventsToReceive == TouchableEventType.Touch)
                 {
                     MixedRealityToolkit.InputSystem?.RaiseOnTouchUpdated(InputSourceParent, Controller, Handedness, touchPosition);
                 }
             }
         }
 
-        public bool TryGetNearGraspPoint(out Vector3 position)
+        private static bool IsObjectPartOfTouchable(GameObject targetObject, NearInteractionTouchable touchable)
+        {
+            return targetObject != null && touchable != null &&
+                (targetObject == touchable.gameObject ||
+                // Descendant game objects are touchable as well. In particular, this is needed to be able to send
+                // touch events to Unity UI control elements.
+                (targetObject.transform != null && touchable.gameObject.transform != null &&
+                targetObject.transform.IsChildOf(touchable.gameObject.transform)));
```

Comment:
Is this something that could possibly run with every frame? If so, do you have a sense of how painful the perf impact is of doing this descendent enumeration on each frame?

Reply:
Yes, it's possibly run once every frame. Just profiled a bit in VR and it took < 0.01ms (< 0.1% frame time) on average. Not sure if caching this result makes sense, if we want to allow the game object hierarchy to change at runtime.

Reply:
👍  sounds good, just wanted to get a gut check here to make sure we considered it.

I suppose in our examples we probably have more simple geometry/hierarchy, so just something for us to keep in mind in the future.